### PR TITLE
[yang-models]: Remove PLY Extensions and change translation code.

### PR DESCRIPTION
**Pull request by praveen-li**
With assumption that TABLE_SEPARATOR and ENTRY_SEPARATOR for configDB is always "|",
   translation from configDB.json to sonicYang.json can be done based on keys specified
   in YANG Lists inside YANG models. So removing extensions is good idea.

Changes:
-- Remove use of regex in Translation code.
-- Remove regex Extensions from YANG models.
-- Improved debugging i.e. log on stdout in case of any Exception from sonic-yang-mgmt,
   so that failed tests can be debugged faster. Also this is good to debug Dynamic
   port breakout issues.
-- Minor Test changes.

#### Why I did it
   With assumption that TABLE_SEPARATOR and ENTRY_SEPARATOR for configDB is always "|",
   translation from configDB.json to sonicYang.json can be done based on keys specified
   in YANG Lists inside YANG models. So removing extensions is good idea.

#### How I did it

-- Remove use of regex in Translation code in sonic_yang_ext.py file.
-- Remove regex Extensions from YANG models.
-- Improved debugging i.e. log on stdout in case of any Exception from sonic-yang-mgmt,
   so that failed tests can be debugged faster. Also this is good to debug Dynamic
   port breakout issues.
-- Minor Test changes.

#### How to verify it

Build time test case already covers this case of translation and reverse translation.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 1]
```diff
@@ -154,25 +159,22 @@ def _cropConfigDB(self, croppedFile=None):
     Input:
     tableKey: Config DB Primary Key, Example tableKey = "Vlan111|2a04:5555:45:6709::1/64"
     keys: key string from YANG list, i.e. 'vlan_name ip-prefix'.
-    regex: A regex to extract keys from tableKeys, good to have it as accurate as possible.
 
     Return:
     KeyDict = {"vlan_name": "Vlan111", "ip-prefix": "2a04:5555:45:6709::1/64"}
     """
-    def _extractKey(self, tableKey, keys, regex):
+    def _extractKey(self, tableKey, keys):
```
Comment (praveen-li) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:166]: This function has the code changes related to translation.

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 2]
```diff
@@ -433,22 +441,24 @@ def _xlateConfigDB(self, xlateFile=None):
     """
     create config DB table key from entry in yang JSON
     """
-    def _createKey(self, entry, regex):
+    def _createKey(self, entry, keys):
```
Comment (praveen-li) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:444]: This function has the code changes related to reverse translation.

**Comment by lguohan** [comment]
can you fix the lgtm alert?

**Comment by praveen-li** [comment]
> can you fix the lgtm alert?

@lguohan fixed lgtm

**Review by lguohan — DISMISSED** [review]

**Comment by praveen-li** [comment]
@lguohan Kindly marge this PR, thanks a lot.

**Review by lguohan — APPROVED** [review]

**Pull request merged**